eng/Packages.Data.props

```diff
@@ -64,14 +64,16 @@
     <PackageReference Update="System.Numerics.Vectors" Version="4.5.0" />
     <PackageReference Update="System.Diagnostics.DiagnosticSource" Version="4.6.0" />
     <PackageReference Update="System.Reflection.TypeExtensions" Version="4.5.1" />
-    <PackageReference Update="System.Threading.Tasks.Extensions" Version="4.5.2" />
+    <PackageReference Update="System.Threading.Tasks.Extensions" Version="4.5.4" />
     <PackageReference Update="System.Threading.Channels" Version="4.6.0" />
     <PackageReference Update="System.Security.Cryptography.ProtectedData" Version="4.5.0" />
     <PackageReference Update="System.Text.Json" Version="4.6.0" />
     <PackageReference Update="System.ValueTuple" Version="4.5.0" />
-    <PackageReference Update="Microsoft.Bcl.AsyncInterfaces" Version="1.0.0" />
+    <PackageReference Update="Microsoft.Bcl.AsyncInterfaces" Version="1.1.1" />
     <PackageReference Update="Microsoft.CSharp" Version="4.5.0" />
-
+    <PackageReference Update="System.Collections.Immutable" Version="1.3.1" />
```

Comment:
We shouldn't expose any immutable collections in the API and they shouldn't be required for implementaitons.

Reply:
Added an ADO item to track this